### PR TITLE
implement - nil datatype - polar null type column

### PR DESCRIPTION
**Pull request by lkarthee**
For addressing issues relating to null datatype.

Its not ready for merging - need code review and some help.

### CODE REVIEW

**Comment by lkarthee** [comment]
@josevalim There are a lot of test failures - i will fix them after review ..

File: lib/explorer/series.ex [hunk 1]
```diff
@@ -919,16 +919,20 @@ defmodule Explorer.Series do
...
+  defp maybe_cast_series({:error, dtype}, _), do: dtype_error("cast/2", dtype, Shared.dtypes())
+
+  defp maybe_cast_series(dtype, %Series{dtype: dtype} = series), do: series
+
+  defp maybe_cast_series(normalised, series), do: apply_series(series, :cast, [normalised])
+
+  @doc type: :element_wise
+  @spec nil_(series :: Series.t()) :: Series.t()
+  def nil_(series) do
+    cast(series, nil)
```
Comment (josevalim) [inline lib/explorer/series.ex:935]: I think I would call it `:null`, so we don't accidentally handle it as `nil`. And i don't think we need a `nil_` function, `cast(series, :null)` would be enough IMO.

Comment (josevalim) [reply lib/explorer/series.ex:935]: Also calling it `:null` would require fewer changes to the code, exactly becaude of stuff like `normalise_dtype` :D

Comment (lkarthee) [reply lib/explorer/series.ex]: Thank you, fixed as :null - that resolved all failing test cases.

**Comment by josevalim** [comment]
Thank you!

Let's add some tests that `to_list` works and returns a list of nils. I assume `to_binary` and `to_iovec` both fail? Do they fail with a decent error message at least?

**Comment by lkarthee** [comment]
Sure I will add test cases for `to_list`. 

Error message for `to_iovec` and `to_binary`.

```
** (ArgumentError) cannot convert series of dtype :null into iovec
    (explorer 0.8.0-dev) lib/explorer/series.ex:769: Explorer.Series.to_iovec/1
    #cell:2wexfollu25fyu43:14: (file)
```

For [1, 2, 3] `to_iovec` fails with : 
```
** (RuntimeError) Generic Error: cannot invoke to_iovec on series with nils
```

How to fix iovec error any pointers ?

**Comment by josevalim** [comment]
Those are all expected and good. So we only need tests for to_list and we are good to go. :)

**Comment by josevalim** [comment]
:green_heart: :blue_heart: :purple_heart: :yellow_heart: :heart:

**Pull request merged**